opencog/query/InitiateSearchCB.cc

```diff
@@ -140,7 +140,7 @@ void InitiateSearchCB::set_pattern(const Variables& vars,
 // start.
 //
 // size_t& depth will be set to the depth of the thinnest constant found.
-// Handle& start will be set to the link containing that constant.
+// Handle& link will be set to the link containing that constant.
```

Comment:
why change this name? This finds and returns the start. The start should have a nice, clear, meaningful name, such as "start". By contrast, almost everything is a "link", we don't need more "links" in the code-base.

Reply:
one more remark about naming conventions: ideally, names should be uniquely grepable.  Thus, for example, in the current code base, I can `cd opencog; grep -r startrm` and I see exactly 6 hits, all in the same file, all for exactly this one and the same variable, and nothing else. If I instead say `cd opencog; grep -r link | wc` I see that the text string `link` occurs in 1991 distinct locations.  This makes debugging pointlessly difficult.  it breaks a basic code-debugging and automation tool.

Reply:
Hmm, OK, but that's really more than ideal, I mean I don't think we need to end with weirder and weirder names just to avoid clashes, nowadays they are code indexing tools that can deal with that (CEDET, rtag, etc, on the Emacs side of things at least).

Reply:
aww com'on -- grep is simple, easy, available everywhere, with no special installation required, no learning curve, no unusual syntax, no special keystrokes. Its just plain-old grep.  Why use some complex tool, when there is an industry-standard simple tool?

Reply:
Original idea was to rename fields to something more readable but I sure "link" is a bad idea because it doesn't mirror sense of the value. Probably I was just confused by the fact that method returns two values and start term is not the actual result while method's name is "find_starter". I would replace it by "start_term".

Not sure grep argument is very relevant here as this is the name of the method parameter and it has a local context you can simply search it within method. Anyway I agree link is misleading name.